### ISSUE TITLE
Please port to safari. 

### ISSUE THREAD

**Issue by drcreek** [corbindavenport/wii-shop-extension#49]

**Comment by EmqsaGit** [comment]
ok... but, i dont know much people that don't INSTANTLLY get another browser when on a mac

**Comment by EmqsaGit** [comment]
and pressed enter too soon, if you do use Safari, that's on you, it's not a BAD browser, but there are many things it doesnt support, aka codecs that are not H.264/5

**Comment by corbindavenport** [comment]
Building extensions for Safari requires joining the [Apple Developer Program](https://developer.apple.com/support/compare-memberships/), which costs $99/year, and you need to use a Mac with Xcode to compile everything. I'm not personally interested in going through that amount of work, and I'm not really sure something with "Wii Shop" in the name that isn't related to Nintendo would pass Apple's guidelines anyway.

I'll leave this issue open if anyone else wants to tackle it and distribute the Wii Shop extension themselves for Safari.

**Comment by qfoxb** [comment]
I think you'd also have to have a mac app associated with an extension for safari, so that'd be just a huge hastle in the first place

**Comment by MisutaaAsriel** [comment]
> I think you'd also have to have a mac app associated with an extension for safari, so that'd be just a huge hastle in the first place

> Building extensions for Safari requires joining the [Apple Developer Program](https://developer.apple.com/support/compare-memberships/), which costs $99/year, and you need to use a Mac with Xcode to compile everything. I'm not personally interested in going through that amount of work, and I'm not really sure something with "Wii Shop" in the name that isn't related to Nintendo would pass Apple's guidelines anyway.
> 
> I'll leave this issue open if anyone else wants to tackle it and distribute the Wii Shop extension themselves for Safari.

**These are actually both false!**

1. XCode ships with a tool to **automatically convert Chrome extensions for Safari, which handles all the app boilerplate.** — Now I couldn't get the extension to *properly* work (no music plays, Notifications are not available to V2 manifests extensions, and the manifest itself required a couple additions to the permissions field). But it *does* install and I can access the menu and everything else. (I've read elsewhere that music not starting may be an issues with Safari as well, where it *readies* but doesn't start depending on how you play it.)
2. **Mac apps, and Safari for macOS extensions, do not require joining the developer program.** — macOS allows for both ad-hoc signing and test-signing applications and extensions with your Apple ID, free of charge (iOS allows the latter *with limitations*). Installing the extension in Safari, when converted, was as simple as enabling unsigned extensions in the Safari developer settings.

**Comment by corbindavenport** [comment]
Right, I meant you can't publish the extensions without paying the fee. There's probably not a lot of people who would be willing to sideload the extension, and I would be doing the extra work of testing in Safari and adding an update check mechanism for that small group of people.

**Comment by MisutaaAsriel** [comment]
> Right, I meant you can't publish the extensions without paying the fee. There's probably not a lot of people who would be willing to sideload the extension, and I would be doing the extra work of testing in Safari and adding an update check mechanism for that small group of people.

That is fair.

I can't really get it working anyways (from what I can tell, Safari is stopping the audio from playing as a user safety measure; Calling `play()` on a regular audio element the user hasn't interacted with is purportedly blocked to prevent sites from circumventing autoplay rules — This doesn't effect audio elements that are set to autoplay, but this extension by nature only plays once it detects a valid site. Not sure how one would get around that beyond autoplaying and immediately pausing...)
